src/barril/units/unit_database.py

```diff
@@ -941,6 +956,18 @@ def GetInfo(self, quantity_type, unit, fix_unknown=False):
                             if info.unit == UNKNOWN_UNIT:
                                 return info
 
+                if fix_legacy:
+                    is_legacy, fixed_unit = fix_unit_if_is_legacy(unit)
+
+                    if is_legacy:
+                        try:
+                            # Common case: unit matches the quantity type registered.
```

Comment:
Hmm what's the uncommon case? The comment suggests there's one 😁

Reply:
Hmm good catch, this comment is a copy & paste mistake. I'll try to refactor to reuse the code.

Reply:
I've refactored this code to use the same code that is used in the beginning of `GetInfo`.